greenmail-core/src/main/java/com/icegreen/greenmail/store/SimpleMessageAttributes.java

```diff
@@ -737,4 +736,9 @@ public static Header create(String[] header) {
             return new Header(header[0]);
         }
     }
+
+    public String escapeHeader(final String text) {
+        return text.replace("\r", "\\\r").replace("\n", "\\\n").replace("\\", "\\\\").replace("\"", "\\\"");

```

Comment:
As discussed, let's use existing JavaMail mime utility for unfoldung and additionally escaping the quote.
As MimeUtility.quote(...) optionally surrounds the result with quotations, you'll also need to remove the quotation when using escapeHeader above.

    return MimeUtility.quote(MimeUtility.unfold(text), javax.mail.internet.HeaderTokenizer.RFC822);